Source/WebCore/dom/Microtasks.h

```diff
@@ -34,6 +34,31 @@ class VM;
 
 namespace WebCore {
 
+class MicrotaskCheckpointScope final {
+    WTF_FORBID_HEAP_ALLOCATION;
+    WTF_MAKE_NONCOPYABLE(MicrotaskCheckpointScope);
+    WTF_MAKE_NONMOVABLE(MicrotaskCheckpointScope);
+public:
+    explicit MicrotaskCheckpointScope(EventLoop& eventLoop)
+    {
+        Ref protectedEventLoop = eventLoop;
+        protectedEventLoop->forEachAssociatedContext([this](auto& context) {
+            m_savedNestingLevels.set(context, context.timerNestingLevel());
+            context.setTimerNestingLevel(0);
+        });
+    }
+
+    ~MicrotaskCheckpointScope()
+    {
+        for (auto contextAndLevel : m_savedNestingLevels) {
+            if (RefPtr context = contextAndLevel.key)
```

Comment:
No need to null-check this. `WeakHashMap` skips over entries where key had already been deleted.